### ISSUE TITLE
Endless loop when `layout` value is `index`

### ISSUE THREAD

**Issue by Tigrov** [yiisoft/view#247]
If specify the path to a layout equal to `index`, the endless loop can occur.

E.g. `$view->withLayout('index')` instead of `$view->withLayout('@layout/index')`.

```php
final class SiteController
{
    public function __construct(private ViewRenderer $viewRenderer)
    {
        $this->viewRenderer = $viewRenderer->withControllerName('site');
    }

    public function index(): ResponseInterface
    {
        return $this->viewRenderer
            ->withLayout('index')
            ->render('index');
    }
}
```

The app try `require index.php` in the endless loop where `index.php` is the app input file.

**Comment by samdark** [comment]
We can:

1. Detect loops in views and error (not sure how since the same view could be legally reused).
2. Throw an error for non-views (there's no good way to tell view from non-view).
3. Throw an error for `index` specifically (not a good idea if base path is changed and view is, ideed, index).

What else?

**Comment by Tigrov** [comment]
4. Check if path to layout has `\/`, if not then error.
5. Add config for default dir for layout.
6. Add to app input file `index.php` check if it already running (can it runs itself again?).

**Comment by samdark** [comment]
Your number 5 sounds reasonable as well as number 4.

**Comment by vjik** [comment]
Usage views in loops may be OK in some cases.

Suggest allow to use for layout only absolute path and throws exception otherwise, because relative path resolved from entry point, and it is unexpected behaviour.

**Comment by samdark** [comment]
> Usage views in loops may be OK in some cases.

Yes. Especially if there's `if()` that stops the loop at some point.

> Suggest allow to use for layout only absolute path and throws exception otherwise, because relative path resolved from entry point, and it is unexpected behaviour.

Makes sense. Let's do it.
